lib/chef/data_collector/config_validation.rb

```diff
@@ -105,7 +110,7 @@ def should_be_enabled?
 
         # validate an output_location file
         def validate_file!(file)
-          open(file, "a") {}
+          File.open(File.expand_path(file), "a") {}
```

Comment:
This seems like the wrong way to test this. Do you know if `File.open` will raise the same exceptions that `open` does? Using `File.exist?` and `File.writable?` would probably be a little better, but I'm not sure what exceptions the last case is trying to catch.

Reply:
@phiggins AFAIK `File.open` raises the same exceptions and `Kernel.open` also works for process and subprocess to open.

I would prefer to use `Chef::Config.path_accessible?` to check the file is writable of not.

I think the last case for sanity purposes only.

Reply:
Yeah this is very hacky code, and I think its just working around windows cases where `File.exist?` / `File.writable?` checks may not be sufficient, so it just tries to open-and-rescue.  I had questions around if it was really necessary and if the cases had been seen in reality and find the whole thing a bit dubious, but didn't know enough to definitively make the call to change this approach.  I kind of left it all alone.